File: tests/file.cpp
#include <vector>
#include <algorithm>
using namespace std;

// Definition for a binary tree node.
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode(int x) : val(x), left(NULL), right(NULL) {}
};

class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        vector<int> sums;
        dfs(root, 0, sums);
        testfunc();
        return max_element(sums.begin(), sums.end()) - sums.begin() + 1;
    }

private:
    void dfs(TreeNode* node, int level, vector<int>& sums) {
        if (level == sums.size()) {
            sums.push_back(node->val);
        } else {
            sums[level] += node->val;
        }
        if (node->left) {
            dfs(node->left, level + 1, sums);
        }
        if (node->right) {
            dfs(node->right, level + 1, sums);
        }
    }

    void testfunc() {
        // Implement your test function here
    }
};

File: tests/file.java
import java.util.*;

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;

    TreeNode(int val) {
        this.val = val;
        this.left = null;
        this.right = null;
    }
}

public class file{
    public static void main(String[] args) {
        TreeNode root = new TreeNode(1);
        root.left = new TreeNode(7);
        root.right = new TreeNode(0);
        root.left.left = new TreeNode(7);
        root.left.right = new TreeNode(-8);
        System.out.println(maxLevelSum(root));
    }

    public static int maxLevelSum(TreeNode root) {
        List<Integer> sums = new ArrayList<>();
        dfs(root, 0, sums);
        return sums.indexOf(Collections.max(sums)) + 1;
    }

    public static void dfs(TreeNode node, int level, List<Integer> sums) {
        if (level == sums.size()) {
            sums.add(node.val);
        } else {
            sums.set(level, sums.get(level) + node.val);
        }
        if (node.left != null) {
            dfs(node.left, level + 1, sums);
        }
        if (node.right != null) {
            dfs(node.right, level + 1, sums);
        }
    }
}